R/parse_expr.R

```diff
@@ -153,6 +153,24 @@ parse_expr_assignment_lhs <- function(lhs, src, call) {
         "E1003", src, call)
     }
 
+    if (special == "dim") {
+      if (length(lhs) < 2) {
+        odin_parse_error(c("Invalid call to dim function; no variables given"),
+                         "E1003", src, call)
+      }
+      lhs <- vcapply(lhs, function(x) {
+        if (!is.symbol(x)) {
+          odin_parse_error("Invalid target '{x}' in dim declaration",
+                           "E1005", src, call)
+        }
+        deparse(x)})[-1]
```

Comment:
```suggestion
      lhs <- vcapply(lhs[-1], function(x) {
        if (!is.symbol(x)) {
          odin_parse_error("Invalid target '{x}' in dim declaration",
                           "E1005", src, call)
        }
        deparse(x)
      })
```

took me a while to work out what the -1 at the end was doing!